dali/test/python/test_operator_readers_numpy.py

```diff
@@ -356,3 +356,81 @@ def test_numpy_reader_alias():
 
         for device in ["cpu", "gpu"] if is_gds_supported() else ["cpu"]:
             yield check_numpy_reader_alias, test_data_root, device
+
+
+def NumpyReaderROIPipeline(file_root, batch_size, device="cpu", file_filter='*.npy',
+                           roi_start=None, rel_roi_start=None, roi_end=None, rel_roi_end=None, roi_shape=None,
+                           rel_roi_shape=None, roi_axes=None, default_axes=[],
+                           num_threads=4, device_id=0):
+    pipe = Pipeline(batch_size=batch_size, num_threads=num_threads, device_id=device_id)
+    with pipe:
+        data = fn.readers.numpy(device=device, file_root=file_root, file_filter=file_filter,
+                                shard_id=0, num_shards=1, cache_header_information=False)
+        roi_data = fn.readers.numpy(device=device, file_root=file_root, file_filter=file_filter,
+                                    roi_start=roi_start, rel_roi_start=rel_roi_start,
+                                    roi_end=roi_end, rel_roi_end=rel_roi_end,
+                                    roi_shape=roi_shape, rel_roi_shape=rel_roi_shape,
+                                    roi_axes=roi_axes,
+                                    shard_id=0, num_shards=1, cache_header_information=False)
+        sliced_data = fn.slice(data, start=roi_start, rel_start=rel_roi_start,
+                               end=roi_end, rel_end=rel_roi_end,
+                               shape=roi_shape, rel_shape=rel_roi_shape,
+                               axes=roi_axes or default_axes)  # Slice has different default (axis_names="WH")
+    pipe.set_outputs(roi_data, sliced_data)
+    return pipe
+
+def check_roi_array(file_root, batch_size, ndim, dtype, device, fortran_order=False, file_filter="*.npy",
+                    roi_start=None, rel_roi_start=None, roi_end=None, rel_roi_end=None, roi_shape=None,
+                    rel_roi_shape=None, roi_axes=None):
+    default_axes = list(range(ndim))
+    pipe = NumpyReaderROIPipeline(file_root=file_root, file_filter=file_filter, device = device, batch_size=batch_size, num_threads=8, device_id=0,
+                                  roi_start=roi_start, rel_roi_start=rel_roi_start, roi_end=roi_end, rel_roi_end=rel_roi_end,
+                                  roi_shape=roi_shape, rel_roi_shape=rel_roi_shape, roi_axes=roi_axes, default_axes=default_axes)
+
+    pipe.build()
+    roi_out, sliced_out = pipe.run()
+    for i in range(batch_size):
+        roi_arr = to_array(roi_out[i])
+        sliced_arr = to_array(sliced_out[i])
+        assert_array_equal(roi_arr, sliced_arr)
+
+def test_numpy_reader_roi():
+    with tempfile.TemporaryDirectory(prefix = gds_data_root) as test_data_root:
```

Comment:
I think you can move it to L24 and don't bother with manual removal of all created files.